tests/robustness/checker/checker.go

```diff
@@ -0,0 +1,275 @@
+// Package checker defines the framework for creating and restoring snapshots
+// with a data integrity check
+package checker
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"time"
+
+	"github.com/kopia/kopia/tests/robustness/snapif"
+	"github.com/kopia/kopia/tests/robustness/snapstore"
+)
+
+// Checker is an object that can take snapshots and restore them, performing
+// a validation for data consistency
+type Checker struct {
+	RestoreDir string
+	snap       snapif.Snapshotter
```

Comment:
If one has not looked in detail at the other packages and type definitions, it is not immediately obvious from the field type or the names what their intended use is.

Regarding the package naming comment in the other PR. While picking a good name is always hard, it'd be good to end up with package and type names that self-describe the fields.

Reply:
renamed the fields